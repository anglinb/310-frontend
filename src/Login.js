import React from 'react';
import {
  Alert,
  StyleSheet,
  KeyboardAvoidingView,
  TextInput,
  Button,
  Text,
  View
} from 'react-native';

export default class Login extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    }
    this.handleButtonPress = this.handleButtonPress.bind(this)
  }

  async handleButtonPress() {
    try {
      let resp = await fetch('http://localhost:3000/authenticate', {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          username: this.state.email,
          password: this.state.password,
        })
      })
      let payload = await resp.json()
      console.log('PAYLOAD', payload)
      if (payload.error) {
        Alert.alert(
          'Whoops!',
          payload.error.msg,
          [
            {text: 'OK', onPress: () => console.log('OK Pressed')},
          ],
          { cancelable: false }
        )
      }
    } catch(error) {
      console.error(error);
    }
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
     <KeyboardAvoidingView style={styles.container}  behavior="padding">
        <TextInput
          style={styles.textInput}
          value={this.state.email}
          onChangeText={(email) => this.setState({email})} />
        <TextInput
          style={styles.textInput}
          value={this.state.password}
          secureTextEntry={true}
          onChangeText={(password) => this.setState({password})} />
        <Button
          title={`Login/Sign Up`}
          onPress={this.handleButtonPress}/>
        </KeyboardAvoidingView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    borderColor: 'red',
    borderWidth: 2,
    width: 200,
  }
});
