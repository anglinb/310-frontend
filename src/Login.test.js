
import React from 'react';

import { shallow } from 'enzyme';
import sinon from 'sinon';

import renderer from 'react-test-renderer';
import Login from './Login';

it('renders without crashing', () => {
  const rendered = renderer.create(
    <Login />
  ).toJSON()
  expect(rendered).toBeTruthy()
  expect(rendered).toMatchSnapshot()
})
