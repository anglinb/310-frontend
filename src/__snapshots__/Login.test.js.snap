// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 40,
        "marginTop": 10,
        "width": 120,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "stretch",
        "display": "flex",
        "paddingTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676E6E",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Email
    </Text>
    <TextInput
      labelText="Email"
      onChangeText={[Function]}
      style={
        Object {
          "backgroundColor": "#D8D8D8",
          "borderRadius": 8,
          "color": "#676E6E",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "fontWeight": "bold",
          "height": 36,
          "padding": 8,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "display": "flex",
        "paddingTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676E6E",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Password
    </Text>
    <TextInput
      labelText="Password"
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#D8D8D8",
          "borderRadius": 8,
          "color": "#676E6E",
          "fontFamily": "Avenir Next",
          "fontSize": 18,
          "fontWeight": "bold",
          "height": 36,
          "padding": 8,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <ExponentLinearGradient
        colors={
          Array [
            4278628033,
            4278632359,
          ]
        }
        end={undefined}
        locations={null}
        start={undefined}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "marginTop": 20,
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "Avenir Next",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Login/Sign Up
        </Text>
      </ExponentLinearGradient>
    </View>
  </View>
</View>
`;
